common/Geometry2d/Pose.hpp

```diff
@@ -0,0 +1,272 @@
+    }
+
+    /**
+     * Implicit conversion to Eigen::Vector3d
+     */
+    operator Eigen::Vector3d() const { return _pose; }
+
+    /**
+     * Calculate a TransformMatrix corresponding to using this pose as the
+     * origin of the coordinate system.
+     */
+    TransformMatrix transform() const {
+        return TransformMatrix(position(), heading());
+    }
+
+    /**
+     * Accessors
+     */
+    Point const position() const {
+        return Point(_pose.template block<2, 1>(0, 0));
+    }
+    double const heading() const { return _pose(2); }
+
+    /**
+     * Operators
+     */
+    Pose operator+(const Pose& other) const {
+        return Pose(_pose + other._pose);
+    }
+    Pose operator-(const Pose& other) const {
+        return Pose(_pose - other._pose);
+    }
+    Pose operator*(double s) const { return Pose(_pose * s); }
+    Pose operator/(double s) const { return Pose(_pose / s); }
+    Pose& operator+=(const Pose& other) {
+        _pose += other._pose;
+        return *this;
+    }
+    Pose& operator-=(const Pose& other) {
+        _pose -= other._pose;
+        return *this;
+    }
+    Pose& operator*=(double s) {
+        _pose *= s;
+        return *this;
+    }
+    Pose& operator/=(double s) {
+        _pose /= s;
+        return *this;
+    }
+
+private:
+    Eigen::Vector3d _pose;
+};
+
+/**
+ * Represents a differential (velocity, acceleration, etc.) in 2d space:
+ * (dx, dy, dh). Uses double-precision floating point numbers.
+ */
+class Twist {
+public:
+    /**
+     * Default constructor - zero-initialize
+     */
+    Twist() = default;
+
+    /**
+     * Linear+angular terms.
+     */
+    Twist(const Point& linear, double angular)
+        : _twist(linear.x(), linear.y(), angular) {}
+
+    /**
+     * Component-wise constructor
+     */
+    Twist(double dx, double dy, double dh) : _twist(dx, dy, dh) {}
+
+    /**
+     * Implicit conversion from Eigen::Vector3d
+     */
+    Twist(const Eigen::Vector3d& other) : _twist(other) {}
+
+    /**
+     * Copy-constructor - default
+     */
+    Twist(const Twist& other) = default;
+
+    /**
+     * Zero
+     */
+    static Twist Zero() { return Twist(Eigen::Vector3d::Zero()); }
+
+    /**
+     * Implicit conversion to Eigen::Vector3d
+     */
+    operator Eigen::Vector3d() const { return _twist; }
+
+    /**
+     * Accessors
+     */
+    Point const linear() const {
+        return Point(_twist.template block<2, 1>(0, 0));
+    }
+    double const angular() const { return _twist(2); }
+
+    /**
+     * Find the resulting pose (delta) of an object starting at the origin and
+     * continuing with constant (world-space) velocity for the specified time
+     * (in seconds).
+     *
+     * Throughout the movement, linear velocity relative to the origin is
+     * constant (but velocity in the pose's reference frame is changing in
+     * direction as the pose rotates)
+     *
+     * Called deltaFixed because it operates fixed to the origin frame.
```

Comment:
I think this should be 'applyFixed' not 'deltaFixed'.

Reply:
On a different note, applyX feels as if it should change the object it's acting on. I can't think of a better name off the top of my head so it's not a big deal, but I figured I'd throw it out there.

Reply:
Yeah that's why I didn't name it apply*. That has an expectation that it mutates.

Reply:
Unless I'm looking at an old version of the code the one on line 190 is applyFixed and the one on line 207 is applyRelative

Reply:
Oops again. Didn't mean to resolve this, I fat fingered. These should have been named delta* instead of apply*.

Reply:
There, fixed.